carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/ExtendedElementLocatorFactory.java

```diff
@@ -23,10 +23,17 @@
 
 public final class ExtendedElementLocatorFactory implements ElementLocatorFactory {
     private final SearchContext searchContext;
+    
+    private boolean isRootElementUsed;
 
-    public ExtendedElementLocatorFactory(SearchContext searchContext) {
+    public ExtendedElementLocatorFactory(SearchContext searchContext, boolean isRootElementUsed) {
         this.searchContext = searchContext;
+        this.isRootElementUsed = isRootElementUsed;
     }
+    
+	public boolean isRootElementUsed() {
+		return isRootElementUsed;
+	}
 
     public ElementLocator createLocator(Field field) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.53.207.202/sonarqube/coding_rules#rule_key=squid%3AS1161)